frontend/src/containers/OnBoarding.js

```diff
@@ -11,34 +11,131 @@ import OnBoardingStepPickCoreContributors from '../components/on_boarding/OnBoar
 import OnBoardingStepCreate from '../components/on_boarding/OnBoardingStepCreate';
 import OnBoardingStepThankYou from '../components/on_boarding/OnBoardingStepThankYou';
 
+import fetchReposFromGitHub from '../actions/github/fetch_repos';
+import fetchContributorsFromGitHub from '../actions/github/fetch_contributors';
+import uploadImage from '../actions/images/upload';
+import notify from '../actions/notification/notify';
+import appendGithubForm from '../actions/form/append_github';
+import createGroupFromGithubRepo from '../actions/groups/create_group_from_github_repo';
+import validateSchema from '../actions/form/validate_schema';
+
+import githubSchema from '../joi_schemas/github';
+
 export class OnBoarding extends Component {
 
-  constructor(props) 
-  {
+  static propTypes = {
+    githubUsername: PropTypes.string,
+    repositories: PropTypes.arrayOf(PropTypes.object),
+    contributors: PropTypes.arrayOf(PropTypes.object),
+    githubForm: PropTypes.object,
+  }
+
+  static defaultProps = {
+    githubUsername: '',
+    repositories: [],
+    contributors: [],
+  }
+
+  constructor(props) {
     super(props);
-    this.state = { step: 0 };
+    this.state = {
+      step: 0
+    };
   }
 
-  render() 
-  {
+  componentWillMount() {
+    const { githubUsername } = this.props;
+    if (githubUsername && this.state.step === 0) {
+      this.setState({step: 2});
+    }
+  }
+
+  componentDidMount() {
+    const {
+      githubUsername,
+      fetchReposFromGitHub,
+      appendGithubForm,
+      params
+    } = this.props;
+
+    if (githubUsername && this.state.step === 2) {
+      appendGithubForm({username: githubUsername, token: params.token});
+      fetchReposFromGitHub(githubUsername);
+    }
+  }
+
+  render() {
     const { step } = this.state;
+    const { contributors, githubForm, repositories } = this.props;
+
     return (
       <div className={`OnBoarding ${step ? '-registering' : ''}`}>
         <Notification />
-        {step !== 5 && <OnBoardingHeader active={Boolean(step)} />}
+        {step !== 5 && <OnBoardingHeader active={Boolean(step)} username={githubForm.attributes.username} />}
         {step === 0 && <OnBoardingHero onClickStart={() => this.setState({step: 1})} />}
-        {step === 1 && <OnBoardingStepConnectGithub onNextStep={() => this.setState({step: 2})} />}
-        {step === 2 && <OnBoardingStepPickRepository onNextStep={() => this.setState({step: 3})} />}
-        {step === 3 && <OnBoardingStepPickCoreContributors onNextStep={() => this.setState({step: 4})} />}
-        {step === 4 && <OnBoardingStepCreate onCreate={() => this.setState({step: 5})}/>}
-        {step === 5 && <OnBoardingStepThankYou onContinue={() => this.setState({step: 0})}/>}
+        {step === 1 && <OnBoardingStepConnectGithub />}
```

Comment:
Tried doing that but it requires the entire statement to be in quotes and makes it hard to read. If you know of another way of doing this in JSX, let me know.